packages/host/tests/integration/commands/show-card-test.gts

```diff
@@ -0,0 +1,465 @@
+      );
+      assert.strictEqual(
+        addedItem.format,
+        'isolated',
+        'Stack item has isolated format',
+      );
+    });
+
+    test('calculates correct stack index based on numberOfStacks', async function (assert) {
+      const cardId = `${testRealmURL}Person/alice`;
+      mockOperatorModeStateService.state = {
+        submode: 'interact',
+        codeSelection: undefined,
+      };
+
+      // Test when numberOfStacks returns 0
+      mockOperatorModeStateService.numberOfStacks = () => 0;
+      await command.execute({ cardId });
+
+      assert.strictEqual(
+        mockOperatorModeStateService.addedStackItems[0].stackIndex,
+        0,
+        'Stack index is 0 when numberOfStacks returns 0',
+      );
+
+      // Reset and test when numberOfStacks returns 3
+      mockOperatorModeStateService.reset();
+      mockOperatorModeStateService.state = {
+        submode: 'interact',
+        codeSelection: undefined,
+      };
+      mockOperatorModeStateService.numberOfStacks = () => 3;
+      await command.execute({ cardId });
+
+      assert.strictEqual(
+        mockOperatorModeStateService.addedStackItems[0].stackIndex,
+        1,
+        'Stack index is 1 when numberOfStacks returns 3 (minimum of 3 and 1)',
+      );
+    });
+  });
+
+  module('code submode', function () {
+    test('updates code path and persists selections when in code submode', async function (assert) {
+      const cardId = `${testRealmURL}Person/alice`;
+      mockOperatorModeStateService.state = {
+        submode: 'code',
+        codeSelection: undefined,
+      };
+
+      await command.execute({ cardId });
+
+      // Verify that playground panel service was called with correct parameters
+      assert.strictEqual(
+        mockPlaygroundPanelService.persistedSelections.length,
+        1,
+        'One selection was persisted',
+      );
+
+      const persistedSelection =
+        mockPlaygroundPanelService.persistedSelections[0];
+      assert.strictEqual(
+        persistedSelection.cardId,
+        cardId,
+        'Persisted selection has correct card ID',
+      );
+      assert.strictEqual(
+        persistedSelection.format,
+        'isolated',
+        'Persisted selection has isolated format',
+      );
+      assert.strictEqual(
+        persistedSelection.stack,
+        undefined,
+        'Persisted selection has undefined stack',
+      );
+
+      // Verify that code path was updated
+      assert.strictEqual(
+        mockOperatorModeStateService.updatedCodePaths.length,
+        1,
+        'Code path was updated',
+      );
+      const updatedPath = mockOperatorModeStateService.updatedCodePaths[0];
+      assert.strictEqual(
+        updatedPath.href,
+        `${testRealmURL}person.gts`,
+        'Code path points to person.gts module',
+      );
+    });
+
+    // Errors thrown in test are not caught by the test framework
+    skip('throws error when card is not found in store', async function (assert) {
```

Comment:
Is this so people reading the tests to understand the command will have more context? Or is there a plan to exercise errors somehow?

Reply:
I'm hoping we're going to be able to figure out how to test card error behavior.